### DISCUSSION TITLE
Dio cancel request is not working

### DISCUSSION THREAD

**Discussion by Clothless** [cfug/dio#2286] [Q&A]
` 

      class DioClient {
          late Dio _dio;
          DioClient() {
            _dio = Dio();
      
            _dio.options = BaseOptions(
                baseUrl: Endpoints.url,
                connectTimeout: Endpoints.connectTimeout,
                receiveTimeout: Endpoints.receiveTimeout,
                responseType: ResponseType.json,
                preserveHeaderCase: true);
        
            _dio.interceptors.addAll([AuthorizationInterceptor(), LoggerInterceptor()]);
          }
        
          // cancal token
          CancelToken cancelRequestToken = CancelToken();
        
          // GET METHOD
          Future<Response> get(
            String url, {
            Map<String, dynamic>? queryParameters,
            Options? options,
            CancelToken? cancelToken,
            ProgressCallback? onReceiveProgress,
          }) async {
            try {
              final Response response = await _dio.get(
                url,
                queryParameters: queryParameters,
                options: options,
                cancelToken: cancelRequestToken,
                onReceiveProgress: onReceiveProgress,
              );
              return response;
            } on DioException {
              rethrow;
            }
          }
        
          // POST METHOD
          Future<Response> post(
            String url, {
            String? token,
            data,
            Map<String, dynamic>? queryParameters,
            Options? options,
            CancelToken? cancelToken,
            ProgressCallback? onSendProgress,
            ProgressCallback? onReceiveProgress,
          }) async {
            try {
              final Response response = await _dio.post(
                url,
                data: data,
                options: options,
                cancelToken: cancelRequestToken,
                onSendProgress: onSendProgress,
                onReceiveProgress: onReceiveProgress,
              );
              return response;
            } catch (e) {
              rethrow;
            }
          }
        
          // PUT METHOD
          Future<Response> put(
            String url, {
            dynamic data,
            Map<String, dynamic>? queryParameters,
            Options? options,
            CancelToken? cancelToken,
            ProgressCallback? onSendProgress,
            ProgressCallback? onReceiveProgress,
          }) async {
            try {
              final Response response = await _dio.put(
                url,
                data: data,
                queryParameters: queryParameters,
                options: options,
                cancelToken: cancelRequestToken,
                onSendProgress: onSendProgress,
                onReceiveProgress: onReceiveProgress,
              );
              return response;
            } catch (e) {
              rethrow;
            }
          }
        
          // DELETE METHOD
          Future<dynamic> delete(
            String url, {
            data,
            Map<String, dynamic>? queryParameters,
            Options? options,
            CancelToken? cancelToken,
          }) async {
            try {
              final Response response = await _dio.delete(
                url,
                data: data,
                queryParameters: queryParameters,
                options: options,
                cancelToken: cancelRequestToken,
              );
              return response.data;
            } catch (e) {
              rethrow;
            }
          }
        
          // Cancel all requests
          void cancelRequests() {
            cancelRequestToken.cancel("cancelled");
            cancelRequestToken.whenCancel.then((value) {
              print("Request Cancelled");
            });
          }
        }

`

Here is the dio service,
So am trying to cancel any request whenever the user navigates to a different screen and am calling the cancelRequest() method to do so, but it's not working, i don't know if am doing something wrong or it's just a bug
And by the way, "Request Cancelled" is printed but the request still continue :)

**Comment by kuhnroyal** [comment]
AFAIK the request can not really be cancelled, just the handling of the response or timeout etc.
